### ISSUE TITLE
tsh.exe is from unknown publisher. 

### ISSUE THREAD

**Issue by benarent** [gravitational/teleport#3012]
**What happened**:
When testing Teleport on Windows, I've had this error message, I had to click `More info` to run Teleport. 

While we don't provide a UI, it's very common for Windows users to directly click an .exe ( I'm not 100% sure on a fresh install if you don't approve it ) 

<img width="448" alt="Windows-defender" src="https://user-images.githubusercontent.com/559288/65530096-2d769c80-deac-11e9-97b0-d65ec69f1715.png">

<img width="470" alt="unknown-publisher" src="https://user-images.githubusercontent.com/559288/65530104-2f406000-deac-11e9-905d-800dee609692.png">

**What you expected to happen**:
Similar to https://github.com/gravitational/teleport/issues/2979  we'll need to sign the package after we compile it. https://docs.microsoft.com/en-us/dotnet/framework/tools/signtool-exe?redirectedfrom=MSDN 

**Environment**:
- Tsh version (use `tsh version`): 4.1
- OS (e.g. from /etc/os-release): Windows

**Comment by webvictim** [comment]
It'll be interesting to see whether this happens when running from the CLI. I don't think I had that issue. Either way, though, we should probably sign the binaries just to make sure.

**Comment by benarent** [comment]
@webvictim Windows isn't super clear for our use case, but let me know if you need help acquiring a [.net license](https://media2.giphy.com/media/7TkSKe6bpCmUBPdebX/giphy.gif). https://stackoverflow.com/questions/84847/how-do-i-create-a-self-signed-certificate-for-code-signing-on-windows

**Comment by webvictim** [comment]
The issue we might have is that we don't build on Windows. I'll see whether anyone else has tried to sign Go Windows binaries when building on Linux.

**Comment by benarent** [comment]
It looks like this is a Windows step, https://stackoverflow.com/questions/51717409/is-there-any-way-to-sign-the-windows-executables-generated-by-the-go-compiler might . I guess we'll need another Windows box to sign this. Maybe do the same setup at our MacOS Apps

**Comment by webvictim** [comment]
Yeah - we don't currently build the binaries on a Windows box so we'd need to explicitly get one to do this.

**Comment by benarent** [comment]
I ran into this again today. It seems we don't need a windows box but do need to purchase an EV Code Signing Cert to pass the Microsoft SmartScreen Filter.  I'll work internally to obtain this certificate and we can use http://manpages.ubuntu.com/manpages/trusty/en/man1/signcode.1.html to sign on a linux box.

**Comment by benarent** [comment]
Thanks to @TravisGary we now have a code signing cert. It's shared in our internal secrets system, with password and `.pfx` file.

**Comment by benarent** [comment]
While recording the Windows video I encountered this again. I tried the `signtool.exe` method and it worked well. 

If we install https://www.mono-project.com/docs/tools+libraries/tools/#security we should be able to run this on Linux

```bash
./signtool.exe sign /f gravitational-code-sigining-cert.pfx /t http://timestamp.digicert.com/scripts/timstamp.dll  /p "PASSWORD" /fd SHA256 tsh.exe
``` 

![image](https://user-images.githubusercontent.com/559288/114224436-e9c87400-9925-11eb-8075-778e6840a5b4.png)

**Comment by benarent** [comment]
I've made more progress on this, just out of interest. 

I was able to sign using https://github.com/mtrojnar/osslsigncode, I had to pass in the password as a text file as it has a comma in it. `osslsigncode` can run on Linux, so we should be able to easily add it our build process. 

```bash
osslsigncode sign -pkcs12  /Users/benarent/tmp/teleport-sign/gravitational-code-sigining-cert.pfx -readpass password.txt  -n "Teleport" -i https://goteleport.com -t http://timestamp.digicert.com -h sha2 -in tsh.exe  -out signtsh.exe
# Succeeded
```

**Comment by russjones** [comment]
@timothyb89 This is complete right? Has it been merged into `master` and backported to `branch/v7`?

**Comment by timothyb89** [comment]
Nearly, it's just getting final approval now: https://github.com/gravitational/teleport/pull/7897

I'll backport it as soon as the PR is merged!

**Comment by russjones** [comment]
@timothyb89 No problem, thank you for the update!
